### DISCUSSION TITLE
Upstream policy

### DISCUSSION THREAD

**Discussion by kurtseifried** [cloudsecurityalliance/gsd-tools#178] [Ideas]
Where we have upstreams, be it software or data or standards, we should have a policy to deal with issues/bugs/etc.

For example, if we need OSV tweaked we should go upstream first, probably filing an issue to discuss the problem (https://github.com/ossf/osv-schema/issues/125#issuecomment-1486206362) and then if approved filing a pull request to actually update the standard (https://github.com/ossf/osv-schema/pull/128).

The same goes for software, and data (e.g. filing bugs against errors in Mozilla and Mageia Linux which were resolved).

The next question is, what do we do if the upstream is non-responsive, slow, or refuses to make the change we ask for?

I would suggest we avoid maintaining forks of things (e.g. having our own version of the OSV schema) where possible, but if necessary we do so (e.g. having an OSV schema with more in-depth requirements to ensure we have more than just an id and modified timestamp. If we do this we will need to both maintain the fork, document it, and ideally try to continue working with the upstream to resolve it. 

Am I missing anything here?

**Comment by jasnow** [comment]
Also SUSE
```
local-scripts/update-suse-urls/add-suse-CVEs-to-GSD.sh
local-scripts/update-suse-urls/get-suse-CVEs.sh
```

**Comment by joshbuker** [comment]
If it is at all humanly possible, I would prefer avoiding a fork of the schema itself.

IMO, unless the need is coming directly from the community and is going unaddressed by the existing formats (e.g. OSV, CVE, CSAF, etc), we should stick purely to the existing schema(s) and relegate additional unsupported metadata into either a namespace or metadata field.

That would still let us capture the data while preventing a splintering of standards which I think we all want to avoid.

**Comment by kurtseifried** [comment]
Something to keep in mind:

There are various degrees of forking. E.g. Adding a single item to the list of allowed values for reference links for example, or updating the schema to require a description is very different than adding/removing/changing an entire type of data in the JSON.

Like all things in life there are tradeoffs, in general working with upstreams is the better way to go, but it doesn't always work.
